plugins/toolbox/src/components/Misc/Diff.tsx

```diff
@@ -0,0 +1,171 @@
+import { DiffEditor } from '@monaco-editor/react';
+import React, { useEffect, useState } from 'react';
+import { Button, FormControl, Grid } from '@material-ui/core';
+import { useStyles } from '../../utils/hooks';
+import * as monaco from 'monaco-editor';
+import { useEffectOnce } from 'react-use';
+import { Select, SelectItem } from '@backstage/core-components';
+
+const options: monaco.editor.IDiffEditorConstructionOptions = {
+    originalEditable: true,
+    diffCodeLens: true,
+    dragAndDrop: true,
+    tabCompletion: 'on',
+    renderSideBySide: true,
+};
+
+const FileUploader = ({
+                          onFileLoad,
+                          id,
+                          buttonText,
+                      }: {
+    onFileLoad: Function;
+    id: string;
+    buttonText: string;
+}) => {
+    return (
+        <>
+            <input
+                type="file"
+                accept="*/*"
+                id={id}
+                hidden
+                onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
+                    if (e?.target?.files && e.target.files.length) {
+                        return onFileLoad(e.target.files[0]);
+                    }
+                    return null;
+                }}
+            />
+            <label htmlFor={id}>
+                <Button variant="contained" color="primary" component="span">
+                    {buttonText}
+                </Button>
+            </label>
+        </>
+    );
+};
+
+function readFileAndSetText(
+    file: File | undefined,
+    setText: (value: ((prevState: string) => string) | string) => void,
+    setLanguage: (value: ((prevState: string) => string) | string) => void,
+    allowedLanguages: MonacoLanguages[],
+) {
+    if (file) {
```

Comment:
I prefer using early exits so

if(!file) {
  return;
}

But works of course this way too ☺️

Reply:
Yes, it's good code pattern! I'll keep consistent with your early exits